### DISCUSSION TITLE
WTF Ethers.js教程计划讨论

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#146] [General]
Etherjs教程计划：
## 入门

1. Hello Vitalik! （`更新到第1讲中，2022.07.31`）
2. Provider类，读取区块链信息（`更新到第2讲中，2022.08.04`）
3. 读取智能合约（`更新到第3讲中，2022.08.05`）
4. 发送ETH（`更新到第4讲中，2022.08.09`）
5. 和智能合约交互（`更新到第5讲中，2022.08.21`）
6. 部署智能合约（`更新到第6讲中，2022.08.24`）
7. 读取事件（`更新到第7讲中，2022.08.26`）
8. 监听事件（`更新到第8讲中，2022.08.28`）
9. 过滤事件（`更新到第9讲中，2022.08.31`）
10. 单位转换（`更新到第10讲中，2022.09.01`）

## 应用
11. callStatic（`更新到第11讲中，2022.09.03`）
12. 验证ERC721合约（`更新到第12讲中，2022.09.05`）
13. 编码calldata（`更新到第13讲中，2022.09.06`）
14. 批量生成钱包（`更新到第14讲中，2022.09.09`）
15. 批量发送（`更新到第15讲中，2022.09.10`）
16. 批量归集（`更新到第16讲中，2022.09.11`）
17. Merkle tree（`更新到第17讲中，2022.09.12`）
18. 数字签名（`更新到第18讲中，2022.09.14`）
19. 操作metamask
20. web3modal
21. rainbowkit
22. wagmi

调用WTF Solidity教程-应用中的合约：比如nft交易那节，可以通过ethersjs自动做成授权和上架，增加教程联动性。

建议可以在这个帖子下留言

**Comment by vecpeng** [comment]
可以介绍下[wagmi](https://wagmi.sh/)?它在etherjs的基础上封装了很多有用的React hooks。

**Reply by AmazingAng** [reply]
好建议

**Comment by EasyChris** [comment]
可以把之前很多WTFSolidity合约在remix里面直接点击测试的方法，写成一个ethersjs脚本去调用。

比如nft交易那节，可以通过ethersjs自动做成授权和上架，增加教程联动性。

**Reply by AmazingAng** [reply]
好建议 更新上去了
